utils/parse_config.py

```diff
@@ -0,0 +1,119 @@
+            "crop": gear_context.config["crop"],
+            "filename": gear_context.config["filename"],
+            "ignore_derived": gear_context.config["ignore_derived"],
+            "ignore_errors": gear_context.config["ignore_errors"],
+            "lossless_scaling": gear_context.config["lossless_scaling"],
+            "merge2d": gear_context.config["merge2d"],
+            "philips_scaling": gear_context.config["philips_scaling"],
+            "single_file_mode": gear_context.config["single_file_mode"],
+            "text_notes_private": gear_context.config["text_notes_private"],
+        }
+
+        if gear_context.config["vol3D"]:
+            gear_args["compress_nifti"] = "3"
+
+        # Anonymization cascade
+        if gear_context.config["pydeface"]:
+            gear_args["anonymize_bids"] = True
+            gear_args["text_notes_private"] = False
+
+    elif FLAG == "pydeface":
+
+        gear_args = {
+            "pydeface_cost": gear_context.config["pydeface_cost"],
+            "template": None,
+            "facemask": None,
+            "pydeface_nocleanup": gear_context.config["pydeface_nocleanup"],
+            "pydeface_verbose": gear_context.config["pydeface_verbose"],
+        }
+
+        if gear_context.get_input_path("pydeface_template"):
+            gear_args["template"] = gear_context.get_input_path("pydeface_template")
+            log.info(f"Found input template for pydeface: {gear_args['template']}")
+        else:
+            log.info("No input template provided for pydeface. Defaults assumed.")
+
+        if gear_context.get_input_path("pydeface_facemask"):
+            gear_args["facemask"] = gear_context.get_input_path("pydeface_facemask")
+            log.info(f"Found input facemask for pydeface: {gear_args['facemask']}")
+        else:
+            log.info("No input facemask provided for pydeface. Defaults assumed.")
+
+    elif FLAG == "resolve":
+
+        gear_args = {
+            "ignore_errors": gear_context.config["ignore_errors"],
+            "retain_sidecar": True,
+            "retain_nifti": True,
+            "pydeface_intermediaries": False,
+            "classification": None,
+            "modality": None,
+        }
+
+        if gear_context.config["bids_sidecar"] == "n":
+            gear_args["retain_sidecar"] = False
+
+        if gear_context.config["bids_sidecar"] == "o":
+            gear_args["retain_nifti"] = False
+
+        if gear_context.config["pydeface_nocleanup"]:
+            gear_args["pydeface_intermediaries"] = True
+
+        try:
+            gear_args["classification"] = gear_context.config["inputs"][
+                "dcm2niix_input"
+            ]["object"]["classification"]
+        except Exception as e:
+            log.info("Cannot determine classification from configuration.")
+            log.debug(e)
+
+        try:
+            gear_args["modality"] = gear_context.config["inputs"]["dcm2niix_input"][
```

Comment:
nit: should not expect a KeyError exception, `gear_context.config.get('inputs', {}).get('dcm2niix_input', {}).get('object', {}).get('modality')` or dotty_dict

Reply:
I'm not sure I follow the issue - could you explain more?

Reply:
@kalebdfischer Could you give me more details on this? I am not sure I follow how that metadata is captured coming off the config.

Reply:
Files won't always have a modality defined at the time the config.json is prepared. Broad exceptions that are part of standard execution are not a great practice. The only exception you'd get here is a KeyError (gear_context.config is a dictionary representation of config.json), so I'd replace `Exception` with `KeyError`. If you access with get notation instead of brackets, you should never have an exception in the first place.

Reply:
Modified modality and classification code to account for these comments.